starter_code/src/movies.js

```diff
@@ -4,17 +4,68 @@
 
 // Get the average of all rates with 2 decimals 
 
+var totalRate = 0;
+
+function ratesAverage (movies) {
+
+var totalRate =  movies.reduce(function(accumulator, movie){
+      return accumulator + movie.rate;
+  }, 0); return  totalRate / movies.length; 
+}
 
 // Get the average of Drama Movies
 
+var dramaRate = movies.filter(function(movie){
+  if (movie.genre == 'Drama'){
+    return movie;
+  }
+});
+
+function dramaMoviesRate (dramaRate) {
+
+var totalRate =  dramaRate.reduce(function(accumulator, movie){
+      return accumulator + movie.rate;
+  }, 0); return  Math.floor((totalRate / dramaRate.length)*100)/100; 
+}
+
 
 // Order by time duration, in growing order
 
 
 // How many movies did STEVEN SPIELBERG
 
+var SPIELBERG = movies.filter(function(movie){
+  if (movie.director == 'Steven Spielberg' && movie.genre == 'Drama'){
+    return movie;
+  } else {  
+  } 
+});
+
+if (SPIELBERG.length === 0){
+  array = undefined;
+};
+
+function howManyMovies(array){
+  if (array.length > 0){
+    return 'Steven Spielberg directed '+ array.length +' drama movies!';
+  } else if (array == undefined){
+    return 'Steven Spielberg directed 0 drama movies!';
+  }
```

Comment:
Try to use the function and include what you need to do inside. If I call the function by itself I will only get two strings back event if I send a different array from the one in movies.